File: fix-jsx-comprehensive.mjs
#!/usr/bin/env node

import { readFileSync, writeFileSync } from 'fs';
import { glob } from 'glob';

console.log('🔧 Fixing comprehensive JSX syntax errors...');

const files = await glob('pages/*.tsx');

for (const file of files) {
  console.log(`Processing ${file}...`);
  
  let content = readFileSync(file, 'utf8');
  let originalContent = content;
  
  // Fix missing commas in destructuring
  content = content.replace(/(\w+)\s+(\w+)\s*=\s*useState/g, '$1, $2 = useState');
  content = content.replace(/(\w+)\s+(\w+)\s*=\s*useMemo/g, '$1, $2 = useMemo');
  content = content.replace(/(\w+)\s+(\w+)\s*=\s*useCallback/g, '$1, $2 = useCallback');
  
  // Fix missing commas in function parameters
  content = content.replace(/\(([^)]+)\s+([^)]+)\)\s*=>/g, '($1, $2) =>');
  
  // Fix missing commas in object properties
  content = content.replace(/(\w+):\s*'([^']+)'\s+(\w+):/g, '$1: \'$2\', $3:');
  content = content.replace(/(\w+):\s*"([^"]+)"\s+(\w+):/g, '$1: "$2", $3:');
  
  // Fix missing commas in array methods
  content = content.replace(/Array\.from\([^,]+,\s*\(([^)]+)\s+([^)]+)\)/g, 'Array.from($1, ($2, $3)');
  
  // Fix missing commas in setTimeout
  content = content.replace(/setTimeout\(([^,]+)\s+(\d+)\)/g, 'setTimeout($1, $2)');
  
  // Fix missing commas in useMemo dependencies
  content = content.replace(/\[([^]]+)\s+([^]]+)\]/g, '[$1, $2]');
  
  // Fix return statement structure
  content = content.replace(/^(\s*)return\s*\(\s*$/gm, '\treturn (');
  content = content.replace(/^(\s*)<>$/gm, '\t\t<>');
  content = content.replace(/^(\s*)<\/>$/gm, '\t\t</>');
  
  // Fix JSX element indentation
  content = content.replace(/^(\s*)<Head>$/gm, '\t\t\t<Head>');
  content = content.replace(/^(\s*)<\/Head>$/gm, '\t\t\t</Head>');
  content = content.replace(/^(\s*)<div/gm, '\t\t\t<div');
  content = content.replace(/^(\s*)<nav/gm, '\t\t\t\t<nav');
  content = content.replace(/^(\s*)<Link/gm, '\t\t\t\t\t<Link');
  
  if (content !== originalContent) {
    writeFileSync(file, content);
    console.log(`✅ Fixed ${file}`);
  } else {
    console.log(`ℹ️  No changes needed for ${file}`);
  }
}

console.log('🎉 Comprehensive JSX syntax fixes completed!');

File: fix_jsx_comprehensive.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_jsx_tags(file_path):
    """Fix JSX tag issues in a file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix the specific pattern: <tag>content</\s*$
        content = re.sub(r'(<[^>]*>[^<]*)\s*</\s*$', r'\1</h1>', content, flags=re.MULTILINE)
        
        # Fix unclosed tags at end of lines
        content = re.sub(r'(<h[1-6][^>]*>[^<]*)\s*$', r'\1</h1>', content, flags=re.MULTILINE)
        content = re.sub(r'(<p[^>]*>[^<]*)\s*$', r'\1</p>', content, flags=re.MULTILINE)
        content = re.sub(r'(<button[^>]*>[^<]*)\s*$', r'\1</button>', content, flags=re.MULTILINE)
        content = re.sub(r'(<div[^>]*>[^<]*)\s*$', r'\1</div>', content, flags=re.MULTILINE)
        
        # Fix specific patterns that are common
        content = re.sub(r'(\s*</\s*$)', '', content, flags=re.MULTILINE)
        
        # Fix missing closing tags for common patterns
        content = re.sub(r'(<h[1-6][^>]*>[^<]*)\s*$', r'\1</h1>', content, flags=re.MULTILINE)
        content = re.sub(r'(<p[^>]*>[^<]*)\s*$', r'\1</p>', content, flags=re.MULTILINE)
        content = re.sub(r'(<button[^>]*>[^<]*)\s*$', r'\1</button>', content, flags=re.MULTILINE)
        content = re.sub(r'(<div[^>]*>[^<]*)\s*$', r'\1</div>', content, flags=re.MULTILINE)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed: {file_path}")
            return True
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix JSX tags in all files"""
    print("Fixing JSX tag issues...")
    
    # Find all TypeScript/JavaScript files
    patterns = ['**/*.tsx', '**/*.ts']
    files_to_fix = []
    
    for pattern in patterns:
        files_to_fix.extend(glob.glob(pattern, recursive=True))
    
    # Filter out node_modules and other directories
    files_to_fix = [f for f in files_to_fix if 'node_modules' not in f and '.git' not in f]
    
    print(f"Found {len(files_to_fix)} files to check")
    
    fixed_count = 0
    for file_path in files_to_fix:
        if fix_jsx_tags(file_path):
            fixed_count += 1
    
    print(f"Fixed {fixed_count} files")

if __name__ == "__main__":
    main()
